runtime/lua/vim/shared.lua

```diff
@@ -6,6 +6,7 @@
 -- or the test suite. (Eventually the test suite will be run in a worker process,
 -- so this wouldn't be a separate case to consider)
 
+---@diagnostic disable-next-line: lowercase-global
```

Comment:
Why don't we add a `diagnostic.globals` opt into `.luarc.json` instead

Reply:
Because this is the only place this is a problem (because it may not be running inside vim). Otherwise luals will handle `vim` correctly.

Reply:
Which is why assume that was intentional (on the global level; pun not intended).

Reply:
Globals is only needed when you don't have metadata